### ISSUE TITLE
Support different endianness

### ISSUE THREAD

**Issue by DefCon42** [Rerumu/Rerubi#11]
Tried running a script compiled from Linux and it doesn't work, most likely due to the endianness of bytes being swapped.

**Comment by Rerumu** [comment]
Will look into it

**Comment by gonzalezjo** [comment]
On Linux, the vast majority of software is compiled for a 64-bit operating system (read: a size_t is 8 bytes instead of 4.) Linux doesn't inherently imply a different endianness, so the issue is probably not that.

**Comment by Rerumu** [comment]
I forgot to eventually look into this; with that said it's probably better if I was provided with an error message from OP. The code is a bit dusty and a rewrite of something so it might be iffy to mess with.

**Comment by DefCon42** [comment]
Linux is big endian by default, while Windows is little endian by default. Rerubi assumes little endian for float and integer reading. 64 bit code works; big endian code does not.

**Comment by DefCon42** [comment]
Lua's compiler uses the default endianness of the platform, which is why the endianness bit is important, and why luac files on Linux platforms will not run on Windows platforms.

**Comment by DefCon42** [comment]
Okay, I'll take that back. Completely wrong there (thanks google). However I don't think that should be the issue.

**Comment by gonzalezjo** [comment]
> Linux is big endian by default, while Windows is little endian by default.

This is false, as you seem to have realized. Endianness is a characteristic of the processor.


The issue is almost definitely related to the size of a size_t.
